src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs

```diff
@@ -545,12 +547,37 @@ private void StoreAggregationResult(TreePage page, Table table, AggregationResul
 
             var relatedPage = indexContext.Transaction.InnerTransaction.LowLevelTransaction.GetPage(pageNumber);
             var relatedTreePage = new TreePage(relatedPage.Pointer, Constants.Storage.PageSize);
-            
-            var message = $"Couldn't find a pre-computed aggregation result for the existing page: {relatedTreePage}.";
+
+            string decompressedDebug = null;
+
+            if (relatedTreePage.IsCompressed)
+            {
+                // let's try to decompress it and check if it's empty
+                // we decompress it for validation purposes only although it's very rare case
+
+                using (var decompressed = tree.DecompressPage(relatedTreePage, skipCache: true))
+                {
+                    if (decompressed.NumberOfEntries == 0)
+                    {
+                        // it's empty so there is no related aggregation result, we can safely skip it
+
+                        return false;
+                    }
+
+                    decompressedDebug = decompressed.ToString();
```

Comment:
@arekpalinski is this a scenario that should never happen but seems to sometime happen?

Reply:
Yes, kind of. It happened on prod. Having a page that is empty after decompressing it is valid case. I believe we'll remove it eventually:

https://github.com/ravendb/ravendb/blob/10196851fc9fafb6206e5eb91ac9b9d62c58e9bd/src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs#L396-L415

It's safe to add this check and skip it here. Although I'm not sure that was the case - couldn't reproduce it locally.